### PR TITLE
Updated to Kotlin 1.3 and Coroutines 1.0.0

### LINKED ISSUE #88: Use Kotlin 1.3
Kotlin 1.3 was recently released, it should be used in Vert.x 3.6 release.

### PR DESCRIPTION
**Pull request by gmariotti**
The only class that should be fixed is `CoroutinesTCKTest.kt`. Not sure if it is because of https://github.com/vert-x3/vertx-codetrans/pull/37 not merged yet or because some code is not generated anymore.

It will close #88.

I won't be able to work on it before the start of next week, someone should probably pick it.

### CODE REVIEW

**Comment by vietj** [comment]
so I need to look at CoroutinesTCKTest.kt ?

File: vertx-lang-kotlin-coroutines/src/test/kotlin/io/vertx/kotlin/coroutines/CoroutineContextTest.kt [hunk 1]
```diff
@@ -5,15 +5,15 @@ import io.vertx.core.DeploymentOptions
 import io.vertx.core.Vertx
 import io.vertx.ext.unit.TestContext
 import io.vertx.ext.unit.junit.VertxUnitRunner
-import kotlinx.coroutines.experimental.CancellationException
```
Comment (vietj) [inline vertx-lang-kotlin-coroutines/src/test/kotlin/io/vertx/kotlin/coroutines/CoroutineContextTest.kt]: great :-)

**Comment by vietj** [comment]
when running this I get:

```
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.3.0:test-compile (>test-compile) on project vertx-lang-kotlin-gen: Compilation failure: Compilation failure: 
[ERROR] Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
[ERROR]     class kotlinx.coroutines.CoroutineDispatcher, unresolved supertypes: kotlin.coroutines.AbstractCoroutineContextElement, kotlin.coroutines.ContinuationInterceptor
[ERROR]     class kotlinx.coroutines.Job, unresolved supertypes: kotlin.coroutines.CoroutineContext.Element
[ERROR] 
[ERROR] /Users/julien/java/vertx-lang-kotlin/vertx-lang-kotlin-gen/src/test/kotlin/io/vertx/lang/kotlin/test/CoroutinesTCKTest.kt:[40,32] Cannot access class 'kotlin.coroutines.CoroutineContext'. Check your module classpath for missing or conflicting dependencies
[ERROR] /Users/julien/java/vertx-lang-kotlin/vertx-lang-kotlin-gen/src/test/kotlin/io/vertx/lang/kotlin/test/CoroutinesTCKTest.kt:[40,32] Type inference failed: fun <T> CoroutineScope.async(context: CoroutineContext = ..., start: CoroutineStart = ..., block: suspend CoroutineScope.() -> T#1 (type parameter of kotlinx.coroutines.async)): Deferred<T#1>
[ERROR] cannot be applied to
[ERROR] receiver: GlobalScope  arguments: (CoroutineDispatcher,suspend CoroutineScope.() -> T#2 (type parameter of io.vertx.lang.kotlin.test.CoroutinesTCKTest.coroutinesTest))
[ERROR] 
[ERROR] /Users/julien/java/vertx-lang-kotlin/vertx-lang-kotlin-gen/src/test/kotlin/io/vertx/lang/kotlin/test/CoroutinesTCKTest.kt:[40,38] Type mismatch: inferred type is CoroutineDispatcher but CoroutineContext was expected
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

**Comment by gmariotti** [comment]
I'm not a maven expert, but my guess is that the `vertx-lang-kotlin-gen` uses the latest `vertx-lang-kotlin-coroutines` snapshot available, which uses Kotlin 1.2. For this reason, it goes on conflict with Kotlin 1.3. But I didn't have enough time to investigate it, to be honest. I just wanted to unblock the migration for someone else.

**Comment by vietj** [comment]
indeed updating and install codetrans fixes the problem.

beside this issue is the PR complete ?

**Comment by vietj** [comment]
can you elaborate what is missing if any ?

**Comment by gmariotti** [comment]
if codetrans fixes the problem, then the migration is completed. I think that the next steps, before the 3.6 release, are to add documentation and tests were they are missing and overall improvements. For example, have you considered migrating the repo to Junit 5?

**Comment by vietj** [comment]
@gmariotti for 3.6.0 we should focus

- improve the documentation
   - to document properly the new coroutine extension generation
   - review existing vertx-core documentation which has specificities
- improve examples in https://github.com/vert-x3/vertx-examples
   - more examples ?
   - migrate examples to use the coroutine extension generation (so replace awaitResult by extension)

in the near future after 3.6, I think it would be great to see if we can come with custom extension for existing vertx modules

junit 5 would be fine I think too but it's more a post 3.6 thing 

3.6 would be the last release of the 3.x branch and after that we focus on 4.0 but we can discuss that on vertx-dev list I think

**Comment by okou19900722** [comment]
vertx-codetrans does not need to depends on kotlin, this is the reason why kotlin conflicts in vertx-lang-kotlin-gen.
 I have removed the kotlin dependency in vertx-codetrans in this [pr](https://github.com/vert-x3/vertx-codetrans/pull/36), and only keep kotlin dependencies in the test environment.

**Comment by vietj** [comment]
good point @okou19900722 I'll update version first though :-)

**Pull request merged**